specification/marketplacecatalog/resource-manager/Search/stable/2022-02-02/search.json

```diff
@@ -0,0 +1,785 @@
+{
+  "x-generator": "NSwag v13.15.10.0 (NJsonSchema v10.6.10.0 (Newtonsoft.Json v13.0.0.0))",
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Marketplace Search API",
+    "version": "2022-02-02"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "produces": [
+    "text/plain",
+    "application/json",
+    "text/json"
+  ],
+  "paths": {
+    "/providers/Microsoft.Marketplace/search": {
+      "get": {
```

Comment:
>get [](http://example.com/codeflow?start=7&length=3)

RPC requires that tenant-level RO APIs are implemented using POST instead of GET. Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/a7dd5c0021d0a84bcad6d4736e3fcc3b319b20af/v1.0/proxy-api-reference.md#request-tenant-wide-operations](https://github.com/Azure/azure-resource-manager-rpc/blob/a7dd5c0021d0a84bcad6d4736e3fcc3b319b20af/v1.0/proxy-api-reference.md#request-tenant-wide-operations).

Reply:
@mentat9 that's not tenant based API. we have several similar API's that come from marketplace that are "GET"

Reply:
Tenant-level, in the sense that it applies across all subscriptions in a tenant, rather than applying to a single subscription (or RG). That is exactly the case covered in the RPC (linked above).

The current Microsoft.Marketplace swagger doesn't document other tenant-level RO APIs using GET (other than operations, which is a special ARM api). The only tenant-level GET documented for your RP is a full resource type supporting GET/PUT/PATCH/DELETE. The search API is a read-only operation which should be POST per the RPC.

Reply:
got it. Changed to POST.